kernel/mmanutils.c

```diff
@@ -99,14 +107,48 @@ static ssize_t _map_file_onto_memory(
 
             memcpy(p, buf, (size_t)n);
             p += n;
+            o += n;
             r -= (size_t)n;
             bytes_read += n;
         }
     }
 
-    /* restore the file position */
-    if (lseek(fd, save_pos, SEEK_SET) == (off_t)-1)
-        goto done;
+    /* get the fd flags */
+    ECHECK(flags = fcntl(fd, F_GETFL));
```

Comment:
We don't record the file open modes in FD table? It seems unnecessary to use a syscall to get the modes while we are inside the kernel.

Reply:
I replaced **fcntl()** with a direct call to **myst_syscall_fcntl()** (the former is a wrapper for calling that latter). The fd flags are not stored in the fd-table so no other choice. The function does an O(1) lookup on the fd and calls fs->fcntl().